tests/api/test_api.py

```diff
@@ -0,0 +1,185 @@
+import json
+import uuid
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+from erica.api.v2.endpoints.est import send_est, get_send_est_job
+from erica.api.v2.endpoints.fsc import request_fsc, get_fsc_request_job, activate_fsc, get_fsc_activation_job, \
+    revocate_fsc, get_fsc_revocation_job
+from erica.api.v2.endpoints.tax import is_valid_tax_number, get_valid_tax_number_job
+from erica.api.v2.responses.model import JobState
+from erica.application.EricaAuftrag.EricaAuftrag import EricaAuftragDto
+from erica.application.JobService.job_service import JobService
+from erica.domain.Shared.EricaAuftrag import RequestType
+from erica.domain.Shared.Status import Status
+from erica.domain.erica_request.erica_request import EricaRequest
+from erica.infrastructure.sqlalchemy.repositories.base_repository import EntityNotFoundError
+from tests.application.job_service.test_job_service import MockEricaRequestRepository, MockRequestController, MockDto, \
+    PickableMock
+from tests.utils import create_unlock_code_request, \
+    create_unlock_code_activation, create_unlock_code_revocation, create_tax_number_validity, create_send_est, \
+    get_job_service_patch, get_erica_request_patch
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("method, data, request_type, endpoint, location",
+                         [(request_fsc, create_unlock_code_request(), RequestType.freischalt_code_request, "fsc",
+                           "fsc/request/"),
+                          (activate_fsc, create_unlock_code_activation(), RequestType.freischalt_code_activate, "fsc",
+                           "fsc/activation/"),
+                          (revocate_fsc, create_unlock_code_revocation(), RequestType.freischalt_code_revocate, "fsc",
+                           "fsc/revocation/"),
+                          (is_valid_tax_number, create_tax_number_validity(), RequestType.check_tax_number, "tax",
+                           "tax_number_validity/"),
+                          (send_est, create_send_est(), RequestType.send_est, "est", "ests/")])
+async def test_if_create_job_returns_location(method, data, request_type, endpoint, location):
+    request_id = uuid.uuid4()
+    job_service = JobService(job_repository=MockEricaRequestRepository(), background_worker=MagicMock(),
+                             request_controller=MockRequestController, payload_type=MockDto, job_method=PickableMock())
+    job_service.add_to_queue = Mock(
+        return_value=EricaAuftragDto(type=request_type, status=Status.new, payload="{}",
+                                     request_id=request_id))
+    with patch(get_job_service_patch(endpoint), MagicMock(return_value=job_service)):
+        response = await method(data)
+        assert response == location + str(request_id)
```

Comment:
I like the parametrize, I think we could have used that more often, honestly. But I don't think I get what this test does. For my understanding (which is low for the new code :D) you mock out everything, thus I don't know what is actually tested mainly because I don't get it from the title of the test. Also, could you rename the parameters as to what is expected? E.g. location -> expected_location or endpoint -> endpoint_to_patch

Side note: Why do you even have to mock this much stuff? Seems like maybe the design isn't optimal? 🙈

Reply:
I am not the biggest fan of parametrize at this point because it makes it not that explicit what is happening. However, if we are able to make it more readable to easily understand what is the input, the mocked parts and the expected output, I would be okay with it, too.


@frederike-ramin The idea is to mock everything below the api level to just make sure that the api acts correct on a unit level. But this might be a sign that we should use a testing module here?

Reply:
I will remodel the parameterized tests so that they are more readable and we then have another look?

Reply:
@frederike-ramin @JulianRoesner I did some minor remodeling (more specific test names, more specific parameter names, test ids). With the test id we get now a more readable test names, so in case only 1 of them failed, we can identify the affected api method more easily. This is the e.g. the output of the 1st parameterized test:
`test_api.py::test_if_post_job_returns_location_with_uuid[request_fsc] PASSED [ 20%]
test_api.py::test_if_post_job_returns_location_with_uuid[activate_fsc] PASSED [ 40%]
test_api.py::test_if_post_job_returns_location_with_uuid[revocate_fsc] PASSED [ 60%]
test_api.py::test_if_post_job_returns_location_with_uuid[is_valid_tax_number] PASSED [ 80%]
test_api.py::test_if_post_job_returns_location_with_uuid[send_est] PASSED [100%]`

I went through the whole pytest-parametrized documentation and there aren't, in my opinion, any further functionalities we could use to make the test even more readable.

WDYT?

Reply:
👍